### ISSUE TITLE
Vault/Consul is unable to make requests to Nomad with enforced TLS 

### ISSUE THREAD

**Issue by johncooler** [hashicorp/nomad#23877]
### Attempted Solutions

Hi, I investigated some source code of Nomad/Vault/Consul and I see that documentation graphs
https://developer.hashicorp.com/nomad/docs/integrations/vault/acl#mutual-tls-in-nomad
https://developer.hashicorp.com/nomad/docs/integrations/consul/acl#mutual-tls-in-nomad
I wonder that we can't provide a clear way for such services to communicate, can we implement it? I could raise PR with such code
There is no any reason to disable SSL checks on both sides

**Comment by tgross** [comment]
@johncooler there's no reason for the Nomad server to authenticate a JWKS request, as it's intentionally wide-open. So the only side that needs TLS authentication is the Vault/Consul side, which we're doing via non-mutual TLS normally. We don't intend to implement this feature.

**Comment by johncooler** [comment]
okay, so how I should setup Nomad&Consul or Nomad&Vault intercommunication without disablin strict SSL mode? I want to use Consul mTLS proxy and Vault features at least

**Comment by tgross** [comment]
@johncooler I think you're getting confused here between the Consul/Vault checking Nomad's JWKS endpoint (for authentication via Workload Identity) and Nomad connecting to Consul/Vault. 

When Nomad connects to Consul or Vault, you'd configure mTLS by including the appropriate configuration values in the `consul` or `vault` configuration blocks. That's [`consul.cert_file`](https://developer.hashicorp.com/nomad/docs/configuration/consul#cert_file), [`consul.key_file`](https://developer.hashicorp.com/nomad/docs/configuration/consul#key_file), [`consul.ca_file`](https://developer.hashicorp.com/nomad/docs/configuration/consul#ca_file) for Consul and [`vault.cert_file`](https://developer.hashicorp.com/nomad/docs/configuration/vault#cert_file), [`vault.key_file`](https://developer.hashicorp.com/nomad/docs/configuration/vault#key_file), [`vault.ca_file`](https://developer.hashicorp.com/nomad/docs/configuration/vault#ca_file) for Vault. 

It's up to you how you want to get the cert material onto the Nomad nodes, but [this tutorial](https://developer.hashicorp.com/nomad/tutorials/integrate-vault/vault-pki-nomad) demonstrates one way using consul-template. If you run into issues with that, I'd recommend checking in on [Discuss](https://discuss.hashicorp.com/c/nomad/28).

**Comment by johncooler** [comment]
> @johncooler I think you're getting confused here between the Consul/Vault checking Nomad's JWKS endpoint (for authentication via Workload Identity) and Nomad connecting to Consul/Vault.
> 
> When Nomad connects to Consul or Vault, you'd configure mTLS by including the appropriate configuration values in the `consul` or `vault` configuration blocks. That's [`consul.cert_file`](https://developer.hashicorp.com/nomad/docs/configuration/consul#cert_file), [`consul.key_file`](https://developer.hashicorp.com/nomad/docs/configuration/consul#key_file), [`consul.ca_file`](https://developer.hashicorp.com/nomad/docs/configuration/consul#ca_file) for Consul and [`vault.cert_file`](https://developer.hashicorp.com/nomad/docs/configuration/vault#cert_file), [`vault.key_file`](https://developer.hashicorp.com/nomad/docs/configuration/vault#key_file), [`vault.ca_file`](https://developer.hashicorp.com/nomad/docs/configuration/vault#ca_file) for Vault.
> 
> It's up to you how you want to get the cert material onto the Nomad nodes, but [this tutorial](https://developer.hashicorp.com/nomad/tutorials/integrate-vault/vault-pki-nomad) demonstrates one way using consul-template. If you run into issues with that, I'd recommend checking in on [Discuss](https://discuss.hashicorp.com/c/nomad/28).

got it, so I should set JWKS one time through insecure channel and then mTLS would be applied for all the communications

LGTM

**Comment by tgross** [comment]
> got it, so I should set JWKS one time through insecure channel and then mTLS would be applied for all the communications

No, I'm saying use non-mutual TLS with [`tls.verify_https_client=false`](https://developer.hashicorp.com/nomad/docs/configuration/tls#verify_https_client) (which is still secured with TLS and ACLs!) for the Nomad HTTP endpoint, and use mTLS for Nomad-to-Consul/Vault communication. If you use non-mutual TLS for the Nomad HTTP endpoint (for JWKS) and then switch it to mTLS you're going to have failures when you rotate the key.

See the [Security Model - Requirements](https://developer.hashicorp.com/nomad/docs/concepts/security#requirements) section where we describe TLS for the HTTP endpoint; this is considered safe under the Nomad security model.
